### DISCUSSION TITLE
Question regarding sending multiple packets back to back

### DISCUSSION THREAD

**Discussion by PNys** [aiortc/aioquic#367] [Q&A]
I am working on a streaming video (in particular a DASH stream) over HTTP/3.
My current goal is to have a simple setup, where the client (simple Chrome website) send a single request to the server (aioquic, obviously) which in turn sends all video segments as fast as possible.
Currently I have a simple routine reading a segment as bytearray and then call send_stream_data and so on for every segment.
However, doing this all the data I am sending is received as a single big packet on the client, after a delay needed to send all the data.
I want all the segments to be sent as individual packets, to have the video stream start without the current delay.
Can anyone give me some advice on how to do this?

I have got this to work using individual streams for every segment, however, I need all this to work on a single stream.

**Comment by jlaine** [comment]
I'm not sure I understand why you want to to send your video over HTTP/3? If you want to transmit packets (aka datagrams).. you should not be using a stream oriented protocol!

**Reply by jdevng** [reply]
Why would you say this? I'm planning to use quic streaming for all sorts of time series data, including video. The only problem I'm aware of is dropped packets, which is addressed at the app layer. The streamed packets can arrive out of order, it doesn't matter. What use case of video transmission would reveal the issue you are concerned about? Thanks!

**Comment by jdevng** [comment]
> Can anyone give me some advice on how to do this?

I'm attempting to do similar, but not with video, just data groups. I'm surprised that it's one packet - I am assuming you are trying to multiplex streams at the frame level?

But overall - I'm having difficulty setting up the 'vision' from RFC9000. I'm sure it's capable, I just can't find what I'm looking for.
